internals/daemon/api_exec.go

```diff
@@ -73,13 +75,29 @@ func v1PostExec(c *Command, req *http.Request, _ *userState) Response {
 		return statusBadRequest("%v", err)
 	}
 
+	// Inherit the pebble daemon environment.
+	environment := make(map[string]string)
+	for _, kv := range os.Environ() {
+		parts := strings.SplitN(kv, "=", 2)
+		key := parts[0]
+		val := ""
+		if len(parts) == 2 {
+			val = parts[1]
+		}
+		environment[key] = val
+	}
+	for key, val := range payload.Environment {
+		// Client env will overwrite existing daemon env, if exists.
```

Comment:
```
// Requested environment takes precedence.
```

This makes it slightly more clear that the environment that takes precedence was explicitly provided in the request. This should be obvious from context, but there were early conversations about making the actual client environment be sent to the daemon transparently, and that's not the case here, so let's avoid that interpretation.

Reply:
That's better, thank you. Added in https://github.com/canonical/pebble/pull/234/commits/00d7495b57e5532de07ecdfd9a49a3c53466f3d0.